# StrigaWGS
 
This repository contains code necessary to reproduce figures and analyses presented in Bellis *et al.* (2022).

- **Fig1_mapxPCoA** contains code and accessory data files to run PCoA, produce map of sequenced samples, and evaluate IBD. 
- **Fig2_kinship** contains code and accessory files to map reads to reference, calculate genetic distances, and produce heatmap of values. 
- **Fig3_germination** contains code and data sheet for analysis of germination rate data.  
- **Fig4_kmers** ontains code, summary tables, and lists of differentiated k-mers (with GST >0.5) as well as code needed for permutation analyses. 
- **Fig5_ANGSD** contains code and data (in `angsd_output` directory) for visualizing allele frequency differencies and expression of the chemocyanin across *S. hermonthica* developmental stages and also to visualize genome-wide patterns of Tajima's D and FST between finger millet-parasitizing and maize-parasitizing individuals of *S. hermonthica* sampled in Homa Bay, Kenya according based on alignments to the *de novo* assembled reference.  

